project-koku/koku
# [COST-4242] process and track metered RHEL by resource id

## Jira Ticket

[COST-4242](https://issues.redhat.com/browse/COST-4242)

## Description

This change will alter the flow for metered RHEL to track last processed time at the resource id level instead of at the provider level. This is to address potential gaps due to when systems report data and the records are included in the cost export. 

## Testing

1. Checkout Branch
2. Restart Koku, optionally with a subs worker to make logs easier to spot `make docker-up-min-with-subs`
3. Create and load test customer data `make create-test-customer` followed by `make load-test-customer-data test_source=AWS`
4. When the data is loading, you should see logs indicating some of the usage accounts do not have relevant records while others do. such as:
```
{'message': 'no relevant resource ids found for usage account 9999999999992.', 'tracing_id': 'd7174c98-e3a7-4882-b9e8-68e982478772', 'schema': 'org1234567', 'provider_type': 'AWS-local', 'provider_uuid': '5e8739ba-6940-448e-a110-aabafccd6ff8'}

THIS LOG IS ON THE DEBUG LEVEL IF YOU DONT SEE IT
{'message': 'identified 23 matching records for metered rhel', 'tracing_id': '18881a5b-0d8c-49ab-b440-eb8ab83b7a07', 'schema': 'org1234567', 'provider_type': 'AWS-local', 'provider_uuid': '2ee5b914-15a0-4459-a1a2-7ec7b4458bf9'}
```
5. Check minio and verify the data is present in the correct location and each file contains the same resource id info as the resource id in the name. There should also not be any data present for the source from earlier that had no relevant records. For example, in the following file I would only expect to see line items for the `i-55555556` resource:
```
koku-bucket/org1234567/AWS/source=fc89be83-1d9d-412c-815d-e176c79de539/date=2023-10-04/subs_ee975faa-5aff-4211-be1e-ced13011eef7_i-55555556_0.csv
```

## Notes

...
